hadoop-crypto/src/test/java/com/palantir/crypto2/hadoop/EncryptedFileSystemTest.java

```diff
@@ -266,4 +266,57 @@ public void testGetCipherAlgorithm_nonDefault() throws IOException, URISyntaxExc
         assertThat(efs.getCipherAlgorithm(), is(cipherAlg));
     }
 
+    @Test
+    public void testDelete_successful() throws IOException {
+        testDeleteFile(false);
+    }
+
+    @Test
+    public void testDelete_recusiveDeleteOnFile() throws IOException {
+        testDeleteFile(true);
+    }
+
+    private void testDeleteFile(boolean recursive) throws IOException {
+        assertTrue(efs.delete(path, recursive));
+
+        assertFalse(efs.exists(path));
+        assertThat(keyStore.get(path.toString()), is(nullValue()));
+    }
+
+    @Test
+    public void testDelete_recursiveDeleteOnDir() throws IOException {
+        Path folderPath = new Path(folder.getRoot().getAbsolutePath());
+        assertTrue(efs.delete(folderPath, true));
+
+        assertFalse(efs.exists(path));
+    }
+
+    @Test
+    public void testDelete_nonRecursiveDeleteOnDir() throws IOException {
+        Path folderPath = new Path(folder.getRoot().getAbsolutePath());
+        try {
+            efs.delete(folderPath, false);
+            fail();
+        } catch (IOException e) {
+            assertThat(e.getMessage(), is(String.format("Directory %s is not empty", folderPath)));
+        }
+    }
+
+    @Test
+    public void testDelete_keyMaterialAlreadyDeleted() throws IOException {
+        keyStore.remove(path.toString());
+        assertThat(keyStore.get(path.toString()), is(nullValue()));
+
+        assertTrue(efs.delete(path, false));
+
+        assertFalse(efs.exists(path));
+    }
+
+    @Test
+    public void testDelete_fileAlreadyDeleted() throws IOException {
+        delegateFs.delete(path, false);
+
+        assertFalse(efs.delete(path, false));
+        assertThat(keyStore.get(path.toString()), is(nullValue()));
```

Comment:
`assertThat(keyStore.get(path.toString()), is(keyMaterial));` above the `efs.delete` call.